src/auth.rs

```diff
@@ -70,3 +111,72 @@ impl Middleware<crate::MiniserveConfig> for Auth {
         Ok(Response::Done(resp))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rstest::rstest_parametrize;
+
+    fn get_hash_func(name: &str) -> impl FnOnce(String) -> Vec<u8> {
```

Comment:
Any chance you can put the type into `test_get_hash` directly without converting it from string? Not required but it would make this a bit neater and less complex.

Reply:
Once again, you selected the first line of the block to comment.

Reply:
Okay, now regarding your question, I had tried yesterday, but it seems that the macro cannot take variable names, it can only take literals (strings, numbers, booleans).

Reply:
What happened is, I tried to pass the function directly, then `cargo test` skips `test_hash_func` as an unused function.

Reply:
Ok, that's too bad then.

Reply:
Maybe we can open an issue in rstest repository?